Pilot Bug Fixes

benloh opened the pull request:
A few notes on the changes...

### 1. Fix data corruption when switching between projects

Fixes #112.

#### To reproduce bug

1. Log in as user1 and create a new prop
2. Leave the prop selected (e.g. the "Edit Entity" button should still be highlighted)
3. Click on "Home" or "Logout"
4. Login if necessary and select a different project
5. Notice the "Edit Entity" button is still visible as if a prop had been selected, but upon opening a different project there shouldn't be any prop selected.
6. Click "Edit Entity" or "Add Prop" and click "Save"
7. The app crashes because you're trying to save the prop on the original project -- most likely the prop id will point to a prop that doesn't exist in the new project.


#### Solution

492b0a3d589f324cf865620f339b84119dc74858 should prevent this problem from happening in the future -- when you close a model or log out, the selection flags are cleared so the "Add Property" button is removed.

03b4ade597af1f2eeccc2d78d600b1f5480d5b68 should allow the project to fail to load gracefully -- so you can at least view the project in question.  The danger here is that you won't really notice the error unless you monitor the RLog.  But theoretically the broken sub prop won't display?

![Image](https://github.com/user-attachments/assets/15e916b4-7989-4812-a669-4e52396a0101)

### 2. Updated comment types for IN

Joshua updated comment types for IN.

### 3. hotfix: Fail gracefully if a comment status update on a removed mech can't be found. 233b1136740bee0cb919f8627f361f5c2416c5e9

Error: `Uncaught Error: can not conform`

This error happens if:
a. user creates a comment on a mech
b. user removes the mech
c. another user opens a project to view it

When the other user opens the project for viewing, it tries to read the comment status update and list the changed comments.  But it encounters a comment that references an object that no longer exists.  The comment system tries to fail gracefully, but `comment-mgr` wasn't dealing with a missing `path` (mech reference), so `PMCData.Mech` call would choke.


### 4. hotfix: Improved RLogs

Research logs will now auto-generate a new file after a period of idle time (currently set to 4 hours).  This addresses the issue of a single research log spanning multiple days especially when running on a Digital Ocean server.  See #119.


### 5. Properly handle click events when Prop and Mech Dialogs are open

Also addresses "Disable ControlBar while PropDialog and MechDialog are open" #126 


### 6. Fix one-way/two-way button display for longer props.

The one-way/two-way mech dialog switch breaks if the prop description wraps to a second line.  This sets a max on the width of props, and fixes the height of the buttons so they don't inadvertently wrap.  Fixes #122.


### 7. Fix Print View cutting off graph and prop/mech list.

The proper fix here is to fully replace the old MUI components.

But here's a quick hack for now, that can hopefully take care of most issues:
* Change `app-container` to scroll rather than behave like a single page app window.
* Show a large fixed width and height for the graph so that the full graph is displayed.

### 8. hotfix: Fix resource error when Teacher opens a model

Fixes #130

A teacher by definition owns many different classrooms.
So when when logged in as a teacher, the admin does not have a classroom set.
The error is happening because we look up resources using the classroomId. So if there is no classroomId, we can't find the resources affiliated with the classroom.

Each model has a groupId associated with it however, and each groupId has a classroom associated with it. So we can use that to look up the classroom for a given model: modelId -> groupId -> classroomId.

When opening a model for a teacher then, we use that implicit classroom.

benloh committed:
Merge branch 'master' into dev

benloh committed:
Merge pull request #108 from theRAPTLab/dev-next

Merge `dev-next` into `dev

Commit:
Updated comments for IN

Commit:
Tweaked comment types to match classroom norms

Commit:
Updated comment types to include P8

benloh committed:
hotfix: Clear selected prop between closing model/logouts.  Fixes #112.

benloh committed:
hotfix: Fail gracefully if a parent prop id is missing.  Addresses #112.

This will at least allow the project to be displayed.

benloh committed:
lint rewrap

benloh committed:
hotfix: Show "<ERROR: property name not found>" instead of easily missed "not found" if a property id can't be found.

benloh committed:
hotfix: Fail gracefully if a comment status update on a removed mech can't be found.

A removed mech might still have a comment status update.  But if the mech is gone, the app chokes trying to find the mech source.

dsriseah committed:
dev-sri-hotfix-117: add fs_lastwrite check to close old file, create new log file if interval exceeds LOG_IDLE_INTERVAL

benloh committed:
Merge pull request #119 from theRAPTLab/dev-sri/hotfix-117

Hotfix: Log File Splitting if IDLE_LOG_INTERVAL Exceeded

benloh committed:
dialog-clicks: Disable MEPanelTools when Prop or Mech Dialog is open.  Fixes #120.

jdanish commented:
Confirmed this seems to be fixed

benloh committed:
dialog-clicks: VM_PropMouseExit needs to publishEvent so hover end is triggered.  Fixes #125.

benloh committed:
dialog-clicks: Suppress controlBar when prop or mech dialog is open. Fixes #120.

benloh committed:
dialog-clicks: Fix one-way/two-way switch button display for long Prop descriptions that wrap

benloh committed:
hack: Show fixed graph width/height for print view.  Needs to be fixed with MUI deprecation. Fixes #123

benloh committed:
Merge pull request #126 from theRAPTLab/dev-bl/dialog-clicks

Disable ControlBar while PropDialog and MechDialog are open

benloh commented:
@jdanish Since this is happening in the middle of a pilot, we'll want to do a ton of QA before accepting this pull request.  Depending on the timing and your comfort level, we might leave this branch open until after the pilot.

It might not hurt to tag 190c9041abe299e59f0643178618cc64750f3641 `Updated comment types to include P8` Feb 11, 2025 at 6:34 AM with the date of pilot, e.g. `Rutgers Pilot Feb 15, 2025` assuming that's what you've been running just in case we need to fall back.

jdanish commented:
Why don't we label that "Indiana Spring Feb 11, Start" and then the 15th hotfix (which we are using now) as "Indiana Spring Feb 15, Mid" ... we start at Rutgers Monday also so they'll be on that. I'll need to branch from the 2nd one to add other comment changes for NJ but they'll be localized to that one file so should be ok.

benloh commented:
Thanks!  Both tagged now.

benloh commented:
Converting to draft so we don't inadvertently merge.  
Merge after study is completed.

jdanish commented:
Sounds good thanks!

On Sat, Feb 22, 2025 at 5:02 PM benloh ***@***.***> wrote:

> Converting to draft so we don't inadvertently merge.
> Merge after study is completed.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/theRAPTLab/meme-2023/pull/118#issuecomment-2676421958>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABKGCKTFKS6IQYNTWUH4XFT2RDXXRAVCNFSM6AAAAABXFGV4EGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDMNZWGQZDCOJVHA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
> [image: benloh]*benloh* left a comment (theRAPTLab/meme-2023#118)
> <https://github.com/theRAPTLab/meme-2023/pull/118#issuecomment-2676421958>
>
> Converting to draft so we don't inadvertently merge.
> Merge after study is completed.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/theRAPTLab/meme-2023/pull/118#issuecomment-2676421958>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABKGCKTFKS6IQYNTWUH4XFT2RDXXRAVCNFSM6AAAAABXFGV4EGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDMNZWGQZDCOJVHA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

benloh committed:
comment-changes: Remove red background for windows resize debugging. Fixes #127.

benloh committed:
hotfix: Fix overflows on short screens causing scrolling issues. Fixes #127.

Addresses bug introduced by 4e5197d05093a8308d4db9013fe9a1ede8f523a7

Commit:
NJ updates

benloh committed:
lint rewrap

benloh committed:
hotfix-implicit-teacher-classroomid: Use the model's groupID to look up the classroom id when logged in as a teacher.  Fixes #130.

Commit:
Reverted to IN comments

benloh committed:
units: lint fix

benloh committed:
units: Add `js-yaml`

benloh committed:
units: lint fix

benloh committed:
units: Add `unitId` to model data

benloh committed:
units: Load units from server

benloh committed:
units: Load client-side unit data from server

benloh committed:
units: Select unit when creating new model.

benloh committed:
units: Load Ratings from Units (not Admin DB)

benloh committed:
units: Load Resources from Units (not Admin DB)

benloh committed:
units: Show unit label in panelbar.

benloh committed:
units: Added safeguards when loading yaml to prevent function injection

benloh committed:
units: Restore resource loading when loading a saved model

benloh committed:
units: Deprecate classroomResources

benloh committed:
units: Deprecate classroomResources in adm-objects too

benloh committed:
units: Add unitId to adm-settings

benloh committed:
units: Load commentTypes from unit

benloh committed:
units: Add Unit data validation

benloh committed:
units: Validate unit folder name

benloh committed:
units: Add units README.md

benloh committed:
units: Add avj (Another JSON Validator) for validation

benloh committed:
units: Add "example" unit.

benloh committed:
units: Default to selected Model's unit

benloh committed:
units: Valid resource types are 'simulation', 'assumption', 'idea', 'report', 'question', 'other'

benloh committed:
units: Remove 'Criteria', 'SentenceStarters', 'ResourcesList', 'RatingsView' from admin interface

benloh committed:
units: Serve unit resources under `/units` rather than `/resources` to allow duplicate filenames and unit folder organization

benloh committed:
units: Fix missing unitId in EVResourceItemDialog url

benloh committed:
units: Fix example unit comment type prompts

benloh committed:
units: Improve legibility of validation error messages

benloh committed:
units: Fix resourceTypesList typo

benloh committed:
units: Remove Unit dialog from Model Select (moved to Admin)

benloh committed:
units: Add unitId to Classroom adm-objects definition

benloh committed:
units: Add Units selector to Classroom Admin Panel

benloh committed:
units: Show selected unit on "Create New Model" button.  Disable if unit is bad.

benloh committed:
units: Clean up debug messages

benloh committed:
units: lint fix

benloh committed:
units: Restore display of Resource ID.  Replace `resource.rsrcId` with `resource.id` for units

benloh committed:
units: Remove Criteria from Help

benloh committed:
units: Disable Classroom / Unit selection if teacher is not selected

benloh committed:
units: Add 'likert' and 'discrete-slider' to comment type validation

benloh committed:
units: Set URTable default sort column using `data`

benloh committed:
units: Show "Units" in Model List Tables.

benloh committed:
units: Add "Reload Units" feature.  Use with caution!

benloh committed:
units: lint fix

benloh committed:
units: Remove resource checks during MoveModel.  Previously needed to account for changed resources when moving between classrooms

benloh committed:
units: Add unit information when cloning a model

benloh committed:
units: Default to first unit if selected unit was removed.  Ensure there is always a valid unit for new models.

benloh committed:
units: Update Classroom panel Unit selector if unit has been removed

benloh committed:
units: Any classroom with a missing unit should auto-select the default unit.

benloh committed:
units: Add units support for Electron.  Required lazy loading ajv because of node dependency loading.

benloh committed:
units: Lint, add doc

benloh committed:
units: Deprecate "resources" folder. Resources are now under individual "/units"

benloh committed:
units: Delay Electron BrowserWindow creation to wait for preload script flush to disk

Reverts 3e4e53d1429b3f83f00b90c91c5b7b6f1586b156

benloh committed:
units: Electron: React 18 deprecated ReactDOM.render() in favor of createRoot()

Fixes `Warning: ReactDOM.render is no longer supported in React 18`

benloh committed:
units: Electron: Hide MZIP Importers with deprecation of MUI

benloh committed:
units: Electron: lint fix

benloh committed:
units: Electron: Deprecate MUI/styled-components.

benloh committed:
units: Electron: remote module is deprecated in Electron 14

Instead, we use window.UR.serverinfo which is set by console-preload.js

benloh committed:
units: Electron: Fix preload script error

Properly fix 3e4e53d1429b3f83f00b90c91c5b7b6f1586b156 and 9e5586a37ede8b932982e4d944437b33e64411c8 and

benloh committed:
units: Electron: lint fix

benloh committed:
units: Electron: Revert lazy load ajv. 3e4e53d1429b3f83f00b90c91c5b7b6f1586b156

benloh committed:
units: Electron: Restore electron display.  MZIP Import by dragging no longer works.

benloh committed:
units: Disable Model list item if unit is not found

benloh committed:
Merge pull request #137 from theRAPTLab/dev-bl/units

Feature: Units
